marlowe-actus/src/Language/Marlowe/ACTUS/Analysis.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE RecordWildCards #-}
+module Language.Marlowe.ACTUS.Analysis(genProjectedCashflows, genZeroRiskAssertions) where
+
+import           Data.Time                                               (Day, fromGregorian)
+import qualified Data.List                                               as L (scanl, tail, zip)
+import           Data.Maybe                                              (fromMaybe)
+import           Data.Sort                                               (sortOn)   
+
+import           Language.Marlowe                                        (Contract(Assert), Value(..), Observation(..))
+import           Language.Marlowe.ACTUS.Definitions.BusinessEvents       (EventType (..), RiskFactors (..))
+import           Language.Marlowe.ACTUS.Definitions.ContractTerms        (ContractTerms(..), Assertion(..))
+import           Language.Marlowe.ACTUS.Definitions.Schedule             (CashFlow (..), ShiftedDay (..),
+                                                                          calculationDay, paymentDay)
+import           Language.Marlowe.ACTUS.MarloweCompat                    (constnt, useval)
+import           Language.Marlowe.ACTUS.Model.INIT.StateInitialization   (inititializeState)
+import           Language.Marlowe.ACTUS.Model.POF.Payoff                 (payoff)
+import           Language.Marlowe.ACTUS.Model.SCHED.ContractSchedule     (schedule)
+import           Language.Marlowe.ACTUS.Model.STF.StateTransition        (stateTransition)
+import           Language.Marlowe.ACTUS.Ops                       (ActusNum (..), YearFractionOps (_y))
+import           Prelude                                          hiding (Fractional, Num, (*), (+), (-), (/))
+
+genProjectedCashflows :: ContractTerms -> [CashFlow]
+genProjectedCashflows terms =
+    let
+        eventTypes   = [IED, MD, RR, IP]
+        analysisDate = fromGregorian 2008 10 22
+        riskFactors = RiskFactors 1.0 1.0 1.0 1.0 analysisDate
+
+        preserveDate e d = (e, d)
+        getSchedule e = fromMaybe [] $ schedule e terms
+        scheduleEvent e = preserveDate e <$> getSchedule e
+        events = concatMap scheduleEvent eventTypes
```

Comment:
Analysis thing is a bit experimental. I’m not proud of this code either - it could be much simpler. Basically it was just ‘fold’ in my initial version.